### PR TITLE
Add update to customers resource

### PR DESCRIPTION
**Pull request by maxheyn**
Adds update() to customers resource

### CODE REVIEW

File: bigc/resources/customers.py [hunk 1]
```diff
@@ -47,6 +47,10 @@ def create(self, first_name: str, last_name: str, email: str, **kwargs) -> dict:
         }
         return self._v3_client.request('POST', '/customers', json=[payload])[0]
 
+    def update(self, customer_group_id: int, data: dict) -> dict:
```
Comment (maxheyn) [inline bigc/resources/customers.py:50]:
```suggestion
    def update(self, customer_id: int, data: dict) -> dict:
```

File: bigc/resources/customers.py [hunk 2]
```diff
@@ -47,6 +47,10 @@ def create(self, first_name: str, last_name: str, email: str, **kwargs) -> dict:
...
+        """Update a specific customer by its ID"""
+        return self._v3_client.request('PUT', f'/customer/{customer_group_id}', json=data)
```
Comment (maxheyn) [inline bigc/resources/customers.py:52]:
```suggestion
        return self._v3_client.request('PUT', f'/customer/{customer_id}', json=data)
```

File: bigc/resources/customers.py [hunk 3]
```diff
@@ -47,6 +47,10 @@ def create(self, first_name: str, last_name: str, email: str, **kwargs) -> dict:
         }
         return self._v3_client.request('POST', '/customers', json=[payload])[0]
 
+    def update(self, customer_id: int, data: dict) -> dict:
+        """Update a specific customer by its ID"""
+        return self._v3_client.request('PUT', f'/customer/{customer_id}', json=data)
```
Comment (maxheyn) [inline bigc/resources/customers.py:52]:
```suggestion
        return self._v3_client.request('PUT', f'/customers/{customer_id}', json=data)
```

File: bigc/resources/customers.py [hunk 4]
```diff
@@ -47,6 +47,10 @@ def create(self, first_name: str, last_name: str, email: str, **kwargs) -> dict:
         }
         return self._v3_client.request('POST', '/customers', json=[payload])[0]
 
+    def update(self, customer_id: int, data: dict) -> dict:
+        """Update a specific customer by its ID"""
+        return self._v3_client.request('PUT', f'/customers/{customer_id}', json=data)
```
Comment (r-thomson) [inline bigc/resources/customers.py:52]: This endpoint doesn't seem to actually exist— the v3 API only has a [bulk update endpoint](https://developer.bigcommerce.com/api-reference/595425896c3ec-update-customers).

File: bigc/resources/customers.py [hunk 5]
```diff
@@ -47,6 +47,10 @@ def create(self, first_name: str, last_name: str, email: str, **kwargs) -> dict:
         }
         return self._v3_client.request('POST', '/customers', json=[payload])[0]
 
+    def update(self, customer_id: int, data: dict) -> dict:
+        """Update a single customer"""
+        return self._v3_client.request('PUT', f'/customers', json={'id': customer_id, **data})[0]
```
Comment (r-thomson) [inline bigc/resources/customers.py:52]:
```suggestion
        payload = {
            'id': customer_id,
            **data,
        }
        return self._v3_client.request('PUT', f'/customers', json=[payload])[0]
```

**Review by r-thomson — CHANGES_REQUESTED** [review]

**Review by r-thomson — APPROVED** [review]

**Pull request merged**